### PR TITLE
Integration test

### PR DESCRIPTION
**Pull request by clin2on3mun**
in this milestone:
- Make sure that the N+1 problem is solved when fetching all posts and their comments for a user.
  - Do not use the bullet gem in this exercise - instead analyze the logs as showed in this [article](https://dev.to/junko911/rails-n-1-queries-and-eager-loading-10eh)
  - In your PR description include the screenshots of your app's logs before and after the fix:
    - Before
    beforeN+1
    - After
  aftern+1

[Integration specs](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#integration-specs)
Use Capybara to write integration tests for each view in your project.

Think on the flow of things and group the tests of page in a way that makes sense. For example, one test would check for the things in the page and another test for clicking links on it.

- User index page:
  - I can see the username of all other users.
  - I can see the profile picture for each user.
  - I can see the number of posts each user has written.
  - When I click on a user, I am redirected to that user's show page.

- user show page:
  - I can see the user's profile picture.
  - I can see the user's username.
  - I can see the number of posts the user has written.
  - I can see the user's bio.
  - I can see the user's first 3 posts.
  - I can see a button that lets me view all of a user's posts.
  - When I click a user's post, it redirects me to that post's show page.
  - When I click to see all posts, it redirects me to the user's post's index page.

-  User post index page:
    - I can see the user's profile picture.
    - I can see the user's username.
    - I can see the number of posts the user has written.
    - I can see a post's title.
    - I can see some of the post's body.
    - I can see the first comments on a post.
    - I can see how many comments a post has.
    - I can see how many likes a post has.
    - I can see a section for pagination if there are more posts than fit on the view.
    - When I click on a post, it redirects me to that post's show page.

-  Post show page:
    - I can see the post's title.
    - I can see who wrote the post.
    - I can see how many comments it has.
    - I can see how many likes it has.
    - I can see the post body.
    - I can see the username of each commentor.
    - I can see the comment each commentor left.

### CODE REVIEW

File: spec/features/user_show_spec.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    it 'the User picture and name' do
+      expect(page.has_xpath?("//img[@src='https://placehold.co/200x133']"))
+      expect(page).to have_content('Munana')
+    end
+    it 'show the content Bio and posts' do
+      expect(page).to have_content('Number of posts: 2')
+      expect(page).to have_content('Bio')
+      expect(page).to have_content('new year')
+      expect(page).to have_content('Get it right and getting it wrong')
+    end
```
Comment (codepantha) [inline spec/features/user_show_spec.rb]: - [ ] Please write tests that asserts you can see the user's first 3 posts

File: spec/features/post_index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      expect(page).to have_content('great job man')
+      expect(page).to have_content('Comment: 1')
+      expect(page).to have_content('Likes: 1')
+      expect(page).to have_content('1')
+    end
+    it "clicking on a post, it redirects me to that post's show page" do
+      visit user_posts_path(@user, page: 1)
+      click_link('new year')
+      expect(page).to have_current_path(user_post_path(@user, @post1))
+    end
```
Comment (codepantha) [inline spec/features/post_index_spec.rb:32]: - [ ] Please write tests that assert that you can see a section for pagination. This is a project requirement 👇 
image

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi team 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Good commit history
✔️ Wrote integration specs for views 📌 
✔️ No linter errors 🟢 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/features/user_show_spec.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    before(:each) { visit user_path(id: @user.id) }
+    it 'the User picture and name' do
+      expect(page.has_xpath?("//img[@src='https://placehold.co/200x133']"))
+      expect(page).to have_content('Munana')
+    end
+    it 'show the content Bio and posts' do
+      expect(page).to have_content('Number of posts: 3')
+      expect(page).to have_content('Bio')
+      expect(page).to have_content('new year')
+      expect(page).to have_content('Get it right and getting it wrong')
```
Comment (Qoosim) [inline spec/features/user_show_spec.rb:21]: [ ] This request changes has not been attended to. Kindly do just that. 
```
    Tests that asserts you can see the user's first 3 posts.
```

### Project Requirement
first-3-posts

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle: 

Hi @team, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


## Highlights
-  Implemented some required changes. :heavy_check_mark: 
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Gitflow. :heavy_check_mark: 



 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Qoosim — APPROVED** [review]
#  *Status: Approved* :tada: :heavy_check_mark: 

Hi @team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
-  Implemented required changes. :heavy_check_mark: 
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Gitflow. :heavy_check_mark: 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**